main/src/ca/uwaterloo/flix/api/lsp/provider/completion/ranker/TypeEnumRanker.scala

```diff
@@ -18,16 +18,44 @@ package ca.uwaterloo.flix.api.lsp.provider.completion.ranker
 
 import ca.uwaterloo.flix.api.lsp.provider.completion.Completion
 import ca.uwaterloo.flix.api.lsp.provider.completion.Completion.TypeEnumCompletion
-import ca.uwaterloo.flix.language.ast.{SourceLocation, Symbol}
+import ca.uwaterloo.flix.language.ast.{SourceKind, SourceLocation, Symbol}
 import ca.uwaterloo.flix.util.collection.MultiMap
 
 object TypeEnumRanker {
 
   /**
     * Find the best type enum completion.
+    *
+    * @param completions the list of completions.
+    * @param enumUses    a map consisting of EnumSym and its SourceLocations
+    * @return            Some(TypeEnumCompletion) if a better completion is possible, else none.
     */
   def findBest(completions: Iterable[Completion], enumUses: MultiMap[Symbol.EnumSym, SourceLocation]): Option[TypeEnumCompletion] = {
-    // TODO
-    None
+    // Remove all none typeEnum completions
+    getTypeEnumCompletions(completions)
+      // Find the typeEnum comp that has 0 Real uses
+      .find(typeEnumComp =>
+        hasNoRealSourceKinds(enumUses(typeEnumComp.enumSym)))
   }
+
+  /**
+    * Returns a list only consisting of typeEnum completions.
+    *
+    * @param completions the list of all possible completions.
+    * @return            a List of TypeEnumCompletion.
+    */
+  private def getTypeEnumCompletions(completions: Iterable[Completion]): Iterable[TypeEnumCompletion] = {
+    completions.collect {
+      case comp: TypeEnumCompletion => comp
+    }
+  }
+
+  /**
+    * Checks if set[SourceLocation] does not contain any SourceKind.Real
+    *
+    * @param set the set of SourceLocations.
+    * @return    true, if none of the SourceKinds are Real, false otherwise.
+    */
+  private def hasNoRealSourceKinds(set: Set[SourceLocation]): Boolean =
```

Comment:
Its generally bad to have functions that are negated. Why not have `hasRealSourceKind` and then use `!` in the call to find?